pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -63,7 +63,7 @@ func (r *Reconciler) create() error {
 				DiskSizeGb:  disk.SizeGb,
 				DiskType:    fmt.Sprintf("zones/%s/diskTypes/%s", zone, disk.Type),
 				Labels:      disk.Labels,
-				SourceImage: disk.Image,
+				SourceImage: fmt.Sprintf("%s%s/global/images/%s", r.computeService.BasePath(), r.projectID, disk.Image),
```

Comment:
Can we assume `r.computeService.BasePath()` will always end with `/`? Will the source image path be still valid even with `%s/%s/global/images/%s`?

Reply:
It does from what I've seen, but we can use the googleapi.ResolveRelative function to be on the safe side, will update.